### PR TITLE
fix: add portal and zindex to date-input

### PR DESCRIPTION
**Pull request by thomasdigby**
When rendering a `DateInput` within a `Dialog`, the `Popover` thats rendered when you click "Open calender" gets cut off within the `Dialog` as it overflows the content area. This update wraps the `Popover` in a `Portal` to allow it to correctly render outside of the bounds of the containing `Dialog`, and also adds a higher z-index to ensure it renders on top of the parent dialog.

Unfortunately, I couldn't add a test for this behaviour as it relies on `toBeVisible` to recognise the bounding box of the `Dialog` and the pixels that are overflowing https://github.com/testing-library/jest-dom#tobevisible

### Before

Screenshot 2022-10-24 at 16 11 21

### After

Screenshot 2022-10-24 at 16 11 34

### CODE REVIEW

File: lib/src/components/date-input/DateInput.tsx [hunk 1]
```diff
@@ -79,42 +80,44 @@ export const DateInput = React.forwardRef<HTMLInputElement, DateInputProps>(
...
-            <Calendar
-              date={date || new Date()}
-              selected={date}
-              onDateSelected={async (date) => {
-                setCalendarOpen(false)
-                await setDate(date.date, false)
-                if (revalidate) revalidate()
+          <Popover.Portal>
+            <Popover.Content
+              css={{ pr: '$sizes$2', zIndex: DIALOG_Z_INDEX }}
```
Comment (thomasdigby) [inline lib/src/components/date-input/DateInput.tsx:85]: Mirroring dialog z-index for now as the popover will be portalled later in the DOM and therefore render on top of the dialog

**Comment by thomasdigby** [comment]
Pinging the schools pod just as an FYI about the incoming fix for this issue @johnclarke-atom @Jdrazong @chris-atomlearning @viniciusvolpe

File: lib/src/components/date-input/__snapshots__/DateInput.test.tsx.snap [hunk 2]
```diff
@@ -58,49 +58,6 @@ exports[`DateInput component renders 1`] = `
     stroke-linejoin: round;
     vertical-align: middle;
   }
-
-  .c-iKnjyA {
```
[the remark this answers is not in the document]
Comment (LimeWub) [reply lib/src/components/date-input/__snapshots__/DateInput.test.tsx.snap:62]: It looks like the css for the `Popover.Content` probably has to do with it not being mounted due to it being in a `Portal` ? Something like that maybe?

**Review by LimeWub — APPROVED** [review]

File: lib/src/components/date-input/__snapshots__/DateInput.test.tsx.snap (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomasdigby) [reply lib/src/components/date-input/__snapshots__/DateInput.test.tsx.snap:62]: Yeah that's what I thought initially but I can't even find a reference in the original snapshot, it's not in the DOM anywhere. Unless I've misunderstood how mounting/rendering works within tests 🤔

**Pull request merged**